metadata/src/main/java/org/apache/kafka/metadata/authorizer/StandardAuthorizer.java

```diff
@@ -96,18 +97,16 @@ public void completeInitialLoad(Exception e) {
     }
 
     @Override
-    public void addAcl(Uuid id, StandardAcl acl) {
-        data.addAcl(id, acl);
-    }
-
-    @Override
-    public void removeAcl(Uuid id) {
-        data.removeAcl(id);
+    public synchronized void loadSnapshot(Map<Uuid, StandardAcl> acls) {
+        data = data.copyWithAllNewAcls(acls.entrySet());
     }
 
     @Override
-    public synchronized void loadSnapshot(Map<Uuid, StandardAcl> acls) {
-        data = data.copyWithNewAcls(acls.entrySet());
+    public synchronized void applyAclChanges(
+        Collection<Map.Entry<Uuid, StandardAcl>> newAcls,
```

Comment:
Could we just take a `Collection<StandardAcl>` since the acl has the ID as a property?

Reply:
StandardAcl doesn't contain an ID

Reply:
However, we could use StandardAclWithId here. That would require another copy in the case where we were loading a snapshot, so I'm not sure if it's worth it.

Reply:
Oh, i must have been looking at StandardAclWithId :), this looks good as-is